design/ceph-deprecate-old-osd-types.md

```diff
@@ -0,0 +1,433 @@
+implementations of the new OSD design, a version of the current binary could be used for stages 1
+through 3 to achieve the minimum viable product more quickly. Splitting the stages can be done
+afterwards as makes sense.
+
+#### Stage 1: Host device inspection (Rook-managed mode only)
+The first stage is an inspection stage that is currently done as part of the existing preparation
+stage. Because all of the host's devices must be inspected, this stage likely needs access to the
+entirety of the host's `/dev` directory. It may also need access to large sections of other
+directories like `/run/udev`.
+
+Similar to the current Rook OSD prepare binary, this proposal calls for a Rook inspection binary
+which will examine the disk hardware, rejecting disks that are already in use by applications other
+than Rook (as per current behavior). The inspection binary stores a host device report in a
+ConfigMap which will be read by the operator. The report need merely contain a list of devices which
+are able to be provisioned for Ceph's use (e.g., `/dev/sdc` or `/dev/disk/by-id/blah`).
+
+Because `ceph-volume` is ultimately responsible for preparing the devices, it is ideal if
+`ceph-volume` does this inspection itself and produces a report detailing the disks it could use
+for later provisioning. The inspection binary should store `ceph-volume`'s report into the
+aforementioned ConfigMap.
+
+As a matter of future security, the pod responsible for inspection should have read-only access to
+`/dev` (and `/run/udev` if needed). This stage should not have write access to any part of the host
+unless it is absolutely required.
+
+#### Stage 2: Process device inspection report (Rook-managed mode only)
+In the second stage, the operator examine the host device inspection report in the ConfigMap. The
+primary inspection processing done would be to filter the list of devices reported based on the
+`deviceFilter` cluster storage setting. This stage can also compare the host `devices` specified in
+the cluster config to the devices found during inspection.
+
+The insertion of stage 2 executed within the operator itself, placed in between stages 1 and 3
+allows for easy security improvements later, and it also helps keep the majority of the Ceph
+configuration logic in the operator itself rather than distributing the logic between multiple
+binaries. Existing logic distribution has contributed to operation which is not always transparent
+or easy to extend.
+
+#### Stage 3: Host device preparation (Rook-managed mode only)
+This stage is one in which `ceph-volume` provisions the devices. Unless a custom Rook binary is
+absolutely necessary, `ceph-volume` can be called directly from the pod's container entrypoint.
+Arguments to `ceph-volume` will be based on the host devices found in stage 1 and processing done
+in stage 2.
+
+For future security, this preparation pod should only need access to the individual disks which
```

Comment:
I don't think this is possible since `ceph-volume` consumes a raw block and spits out an LV, I'm not 100% sure but I believe the preparation step requires access to that LV.